imap/jmap_mail.c

```diff
@@ -9048,16 +9082,15 @@ static json_t *_header_from_raw(json_t *jraw,
                                 const char *header_name,
                                 enum header_form form __attribute__((unused)))
 {
-    /* Verbatim use header value in raw form */
-    if (json_is_string(jraw)) {
-        json_t *jheader = json_pack("{s:s s:O s:s}",
-                "name", header_name, "value", jraw, "prop", prop_name);
-        return jheader;
-    }
-    else {
+    if (!json_is_string(jraw)) {
         jmap_parser_invalid(parser, prop_name);
         return NULL;
     }
+
+    struct buf val = BUF_INITIALIZER;
+    buf_setmap(&val, json_string_value(jraw),
+            json_string_length(jraw));
```

Comment:
I wonder if we can avoid a memcpy() by using buf_init_ro() or buf_init_ro_cstr() instead

Reply:
Done, thanks